src/project-generator.ts

```diff
@@ -105,13 +105,23 @@ export class ProjectGenerator {
         return convertedCode;
     }
 
-    private async getNewAppManifestSha(appName: string, vspecPath: string, dataPoints: DataPointDefinition[]): Promise<string> {
+    private async getNewAppManifestSha(appName: string, vspecPath: string, dataPoints: any[]): Promise<string> {
         const appManifestContentData = await this.gitRequestHandler.getFileContentData(APP_MANIFEST_PATH);
-        let decodedAppManifestContent: AppManifest[] = JSON.parse(decode(appManifestContentData));
+        let decodedAppManifestContent = JSON.parse(decode(appManifestContentData));
 
-        decodedAppManifestContent[0].name = appName.toLowerCase();
-        decodedAppManifestContent[0].vehicleModel.src = vspecPath;
-        decodedAppManifestContent[0].vehicleModel.datapoints = dataPoints;
+        if (decodedAppManifestContent instanceof Array) {
```

Comment:
Better to hide this detail in a method `isLegacyAppManifest(decodedAppManifestContent)`